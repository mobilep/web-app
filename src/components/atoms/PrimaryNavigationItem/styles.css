.PrimaryNavigationItem {
	position: relative;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	min-width: 50px;
	max-width: 200px;
	height: var(--primary-navigation-height-desktop);
	color: var(--white);
	font-size: 9px;
	line-height: normal;
	font-weight: var(--semiBold);
	transition: all .2s ease-in-out;
	border-radius: 0;
	padding: 10px 5px;
	border: 0;
	border-bottom: 7px solid transparent;
}

.PrimaryNavigationItem:hover {
	border-bottom-color: var(--pinkRed);
	background-color: var(--tomatoRed);
	text-decoration: none;
}

.PrimaryNavigationItem-active,
.PrimaryNavigationItem-active:hover {
	border-bottom-color: #fff;
	background-color: #fff;
	color: #f3423c;
}

.PrimaryNavigationItem-active .IconWithCounter-counter {
	background-color: var(--tomatoRed);
	color: #fff;
}

.PrimaryNavigationItem > svg,
.PrimaryNavigationItem .IconWithCounter {
	width: 26px;
	height: 26px;
	margin-bottom: 3px;
}

@media (max-width: 768px) {
	.PrimaryNavigationItem {
		height: 53px;
		padding: 0;
		border-bottom: none;
	}

	.PrimaryNavigationItem,
	.PrimaryNavigationItem:hover {
		background-color: var(--white);
		color: var(--grey3);
	}

	.PrimaryNavigationItem-active,
	.PrimaryNavigationItem-active:hover {
		color: var(--tomatoRed);
	}
}
