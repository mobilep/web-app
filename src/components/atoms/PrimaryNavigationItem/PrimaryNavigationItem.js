import React from 'react';
import { Button } from '..';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './styles.css';

const getClassName = (className) => classNames('PrimaryNavigationItem', { [className]: className });

const PrimaryNavigationItem = ({ children, className, ...rest }) => (
	<Button
		{ ...rest }
		className={getClassName(className)}
		activeClassName='PrimaryNavigationItem-active'
	>
		{children}
	</Button>
);

PrimaryNavigationItem.propTypes = {
	children: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.element]),
	className: PropTypes.string,
};

export default PrimaryNavigationItem;
