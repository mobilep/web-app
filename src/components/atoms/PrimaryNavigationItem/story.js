import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';
import PrimaryNavigationItem from './PrimaryNavigationItem';
import StoryRouter from 'storybook-react-router';
import { InboxIcon } from '../../../icons';

storiesOf('1. Atoms / PrimaryNavigationItem', module)
	.addDecorator(StoryRouter())
	.add('default',
		() => (<div style={{ backgroundColor: 'red', width: '64px' }}>
			<PrimaryNavigationItem to={text('to', '/someUrl')}>
				<InboxIcon />
				<span>{text('text', 'INBOX')}</span>
			</PrimaryNavigationItem>
		</div>));
