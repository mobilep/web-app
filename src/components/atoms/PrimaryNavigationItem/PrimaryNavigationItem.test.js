import React from 'react';
import { shallow } from 'enzyme';

import PrimaryNavigationItem from './PrimaryNavigationItem';

describe('Atom - PrimaryNavigationItem', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<PrimaryNavigationItem>
				Item
			</PrimaryNavigationItem>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
