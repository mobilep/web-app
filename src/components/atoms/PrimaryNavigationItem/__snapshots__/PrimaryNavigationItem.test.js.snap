// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - PrimaryNavigationItem Should render 1`] = `
<Button
  activeClassName="PrimaryNavigationItem-active"
  className="PrimaryNavigationItem"
>
  Item
</Button>
`;
